third_party/move/tools/move-spec-test/src/report.rs

```diff
@@ -0,0 +1,202 @@
+use serde::Serialize;
+use std::collections::HashMap;
+use std::path::PathBuf;
+use tabled::{builder::Builder, settings::Style};
+
+/// This struct represents a report of the specification testing.
+/// It contains the list of entries, where each entry is a file and the number of mutants tested
+/// and killed in that file (in form of a `ReportEntry` structure).
+#[derive(Debug, Serialize)]
+pub struct Report {
+    /// The list of entries in the report.
+    files: HashMap<PathBuf, MutantStats>,
+}
+
+impl Report {
+    /// Creates a new report.
+    pub fn new() -> Self {
+        Self {
+            files: HashMap::new(),
+        }
+    }
+
+    /// Adds an entry to the report.
+    pub fn add_entry(&mut self, path: PathBuf, entry: MutantStats) {
+        self.files.insert(path, entry);
+    }
+
+    /// Returns `true` if the report contains an entry for the given path.
+    pub fn contains(&self, path: &PathBuf) -> bool {
+        self.files.contains_key(path)
+    }
+
+    /// Increments the number of mutants tested for the given path.
+    /// If the path is not in the report, it adds it with the number of mutants tested set to 1.
+    pub fn increment_mutants_tested(&mut self, path: &PathBuf) {
+        if self.contains(path) {
+            let entry = self.files.get_mut(path).unwrap();
+            entry.tested += 1;
+        } else {
+            self.add_entry(
+                path.clone(),
+                MutantStats {
+                    tested: 1,
+                    killed: 0,
+                },
+            );
+        }
+    }
+
+    /// Increments the number of mutants killed for the given path.
+    /// If the path is not in the report, it adds it with the number of mutants tested and killed set to 1.
+    pub fn increment_mutants_killed(&mut self, path: &PathBuf) {
+        if self.contains(path) {
+            let entry = self.files.get_mut(path).unwrap();
+            entry.killed += 1;
+        } else {
+            self.add_entry(
+                path.clone(),
+                MutantStats {
+                    tested: 1,
+                    killed: 1,
+                },
+            );
+        }
+    }
+
+    /// Returns the number of mutants tested.
+    pub fn mutants_tested(&self) -> u64 {
+        self.files.values().map(|entry| entry.tested).sum()
+    }
+
+    /// Returns the number of mutants killed.
+    pub fn mutants_killed(&self) -> u64 {
+        self.files.values().map(|entry| entry.killed).sum()
+    }
+
+    /// Save the report to a JSON file.
+    /// The file is created if it does not exist, otherwise it is overwritten.
+    pub fn save_to_json_file(&self, path: &PathBuf) -> anyhow::Result<()> {
+        let file = std::fs::File::create(path)?;
+        serde_json::to_writer_pretty(file, self)?;
+        Ok(())
```

Comment:
Would this work? (Resolve the comment if not)
```suggestion
        serde_json::to_writer_pretty(file, self)
```

Reply:
No.
```
   --> third_party/move/tools/move-spec-test/src/report.rs:51:9
    |
49  |     pub fn save_to_json_file(&self, path: &PathBuf) -> anyhow::Result<()> {
    |                                                        ------------------ expected `Result<(), anyhow::Error>` because of return type
50  |         let file = std::fs::File::create(path)?;
51  |         serde_json::to_writer_pretty(file, self)
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `anyhow::Error`, found `serde_json::Error`
```

But I can change to this:
`Ok(serde_json::to_writer_pretty(file, self)?)`